bot/bot.py

```diff
@@ -44,9 +45,27 @@ def __init__(self, *args, redis_session: RedisSession, **kwargs):
             # Since statsd is UDP, there are no errors for sending to a down port.
             # For this reason, setting the statsd host to 127.0.0.1 for development
             # will effectively disable stats.
-            statsd_url = "127.0.0.1"
+            statsd_url = LOCALHOST
+
+        try:
+            self.stats = AsyncStatsClient(self.loop, statsd_url, 8125, prefix="bot")
+        except socket.gaierror as socket_error:
+            self.stats = AsyncStatsClient(self.loop, LOCALHOST)
+            self.loop.call_later(30, self.retry_statsd_connection, statsd_url)
+            log.warning(f"Statsd client failed to instantiate with error:\n{socket_error}")
```

Comment:
This feels redundant. It can be replaced with a call to `retry_statsd_connection` if you do `self.stats = AsyncStatsClient(self.loop, LOCALHOST)` inside the function. Maybe after an `if attempt == 1`. Or you could set it before you call the function.